lib/collection/src/shards/resharding/driver.rs

```diff
@@ -531,14 +531,26 @@ async fn migrate_local(
 
     let progress = Arc::new(Mutex::new(TransferTaskProgress::new()));
     transfer_resharding_stream_records(
-        shard_holder,
+        Arc::clone(&shard_holder),
```

Comment:
```suggestion
        shard_holder.clone(),
```

Reply:
Good call. There are other bits in code that'll recover it. But it won't hurt to be safe here as well.

I'll add it and will merge after.

Reply:
0e4b2a794512d2397a4d214eab9782e1d464d542